presto-main/src/main/java/com/facebook/presto/sql/rewrite/ExplainRewrite.java

```diff
@@ -100,25 +101,34 @@ protected Node visitExplain(Explain node, Void context)
                 }
             }
 
-            String plan = getQueryPlan(node, planType, planFormat);
-
-            return singleValueQuery("Query Plan", plan);
+            return getQueryPlan(node, planType, planFormat);
         }
 
-        private String getQueryPlan(Explain node, ExplainType.Type planType, ExplainFormat.Type planFormat)
+        private Node getQueryPlan(Explain node, ExplainType.Type planType, ExplainFormat.Type planFormat)
                 throws IllegalArgumentException
         {
             Statement wrappedStatement = node.getStatement();
             Statement statement = unwrapExecuteStatement(wrappedStatement, parser, session);
             List<Expression> parameters = wrappedStatement instanceof Execute ? ((Execute) wrappedStatement).getParameters() : emptyList();
             validateParameters(statement, parameters);
+
+            if (planType == VALIDATE) {
+                queryExplainer.get().analyze(session, statement, parameters);
+                return singleValueQuery("Valid", Boolean.valueOf(true).toString());
```

Comment:
This still returns a `varchar`. You'll need another version of `singleValueQuery` that returns a boolean.